experimental/yarpl/include/yarpl/Flowable.h

```diff
@@ -2,48 +2,87 @@
 
 #pragma once
 
-#include <functional>
 #include <memory>
-#include "yarpl/Flowable_Subscriber.h"
+#include <type_traits>
+#include <utility>
+#include "yarpl/flowable/sources/Flowable_RangeSubscription.h"
+
+#include "reactivestreams/ReactiveStreams.h"
+#include "yarpl/utils/type_traits.h"
 
 namespace yarpl {
 namespace flowable {
 
+// using reactivestream_yarpl to not conflict with the other reactivestreams
+
 template <typename T>
-class Flowable {
+class Flowable : public reactivestreams_yarpl::Publisher<T>,
+                 public std::enable_shared_from_this<Flowable<T>> {
+  using Subscriber = reactivestreams_yarpl::Subscriber<T>;
+  using Subscription = reactivestreams_yarpl::Subscription;
+
  public:
-  explicit Flowable(
-      std::function<void(std::unique_ptr<Subscriber<T>>)> onSubscribe);
+  template <
+      typename F,
+      typename = typename std::enable_if<
+          std::is_callable<F(std::unique_ptr<Subscriber>), void>::value>::type>
+  static std::unique_ptr<Flowable> create(F&& onSubscribeFunc) {
```

Comment:
repeating same mistakes like in vjn's diff
You are implementing perfect forwarding here incorrectly: http://stackoverflow.com/questions/6829241/perfect-forwarding-whats-it-all-about